### PR TITLE
chore: Change MPL branch, remove PYTHONPATH workarounds

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
*Issue #, if available:*

*Description of changes:*

* Use the "V2" MPL built on Dafny 4.7
* Remove Pythonpath hacks
* Swallow any expected exceptions in test vectors (cleaner test logs)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -25,11 +25,7 @@ jobs:
       matrix:
         os:
           - ubuntu-latest
-          # Windows fails due to "No module named 'Wrappers'"
-          # This SHOULD be fixed once Dafny generates fully-qualified import statements
-          # (i.e. doo files, per-package module names)
-          # Disable for now
-          # - windows-latest
+          - windows-latest
```
Comment (lucasmcdonald3) [inline .github/workflows/ci_tests.yaml:28]:
```suggestion
          - windows-latest 
```

File: .github/workflows/ci_tests.yaml [hunk 2]
```diff
@@ -25,11 +25,7 @@ jobs:
...
```
Comment (lucasmcdonald3) [inline .github/workflows/ci_tests.yaml:28]:
```suggestion
          - windows-latest
```

File: .github/workflows/ci_tests.yaml (hunk 1 above)
Comment (lucasmcdonald3) [inline .github/workflows/ci_tests.yaml]:
```suggestion
          - windows-latest 
```

File: .github/workflows/ci_tests.yaml (hunk 2 above)
Comment (lucasmcdonald3) [reply .github/workflows/ci_tests.yaml:28]:
```suggestion
          - windows-latest
```

Comment (lucasmcdonald3) [reply .github/workflows/ci_tests.yaml:28]:
```suggestion
          - windows-latest   
```

File: examples/src/branch_key_id_supplier_example.py [hunk 3]
```diff
@@ -25,15 +25,17 @@ def get_branch_key_id(
         """Returns branch key ID from the tenant ID in input's encryption context."""
         encryption_context: Dict[str, str] = param.encryption_context
 
-        if b"tenant" not in encryption_context:
+        print(f"{encryption_context=}")
+
```
Comment (lucasmcdonald3) [inline examples/src/branch_key_id_supplier_example.py:29]:
```suggestion
```

File: .github/workflows/ci_tests.yaml [hunk 4]
```diff
@@ -25,11 +25,7 @@ jobs:
...
```
Comment (lucasmcdonald3) [inline .github/workflows/ci_tests.yaml:28]:
```suggestion
          - windows-latest
```

**Review by RitvikKapila — COMMENTED** [review]
This looks good to me. Just one thing, I think you missed one of the examples: https://github.com/aws/aws-encryption-sdk-python/blob/lucmcdon/mpl-v2/examples/src/default_cryptographic_materials_manager_example.py for removing the sys.path.append commands.

**Review by RitvikKapila — APPROVED** [review]
LGTM

**Pull request merged**